src/systemd.c

```diff
@@ -0,0 +1,512 @@
+    },
+    {
+        .accounting_flag = "IOAccounting",
+        .metrics =
+            (systemd_metric[]){
+                {
+                    .name = "IOReadBytes",
+                    .dbus_type = "t",
+                    .collectd_type = METRIC_TYPE_COUNTER,
+                },
+                {
+                    .name = "IOReadOperations",
+                    .dbus_type = "t",
+                    .collectd_type = METRIC_TYPE_COUNTER,
+                },
+                {
+                    .name = "IOWriteBytes",
+                    .dbus_type = "t",
+                    .collectd_type = METRIC_TYPE_COUNTER,
+                },
+                {
+                    .name = "IOWriteOperations",
+                    .dbus_type = "t",
+                    .collectd_type = METRIC_TYPE_COUNTER,
+                },
+                {.collectd_type = METRIC_TYPE_UNTYPED},
+            },
+    },
+    {
+        .accounting_flag = "CPUAccounting",
+        .metrics =
+            (systemd_metric[]){
+                {
+                    .name = "CPUUsageNSec",
+                    .dbus_type = "t",
+                    .collectd_type = METRIC_TYPE_COUNTER,
+                },
+                {.collectd_type = METRIC_TYPE_UNTYPED},
+            },
+    },
+    {
+        .accounting_flag = "IPAccounting",
+        .metrics =
+            (systemd_metric[]){
+                {
+                    .name = "IPEgressBytes",
+                    .dbus_type = "t",
+                    .collectd_type = METRIC_TYPE_COUNTER,
+                },
+                {
+                    .name = "IPEgressPackets",
+                    .dbus_type = "t",
+                    .collectd_type = METRIC_TYPE_COUNTER,
+                },
+                {
+                    .name = "IPIngressBytes",
+                    .dbus_type = "t",
+                    .collectd_type = METRIC_TYPE_COUNTER,
+                },
+                {
+                    .name = "IPIngressPackets",
+                    .dbus_type = "t",
+                    .collectd_type = METRIC_TYPE_COUNTER,
+                },
+                {.collectd_type = METRIC_TYPE_UNTYPED},
+            },
+    },
+    {
+        .accounting_flag = "TasksAccounting",
+        .metrics =
+            (systemd_metric[]){
+                {
+                    .name = "TasksCurrent",
+                    .dbus_type = "t",
+                    .collectd_type = METRIC_TYPE_GAUGE,
+                },
+                {.collectd_type = METRIC_TYPE_UNTYPED},
+            },
+    },
+    {
+        .accounting_flag = NULL,
+        .metrics =
+            (systemd_metric[]){
+                {
+                    .name = "NRestarts",
+                    .dbus_type = "u",
+                    .collectd_type = METRIC_TYPE_COUNTER,
+                },
+                {.collectd_type = METRIC_TYPE_UNTYPED},
+            },
+    },
+};
+
+static systemd_metric_group slice_groups[] = {
```

Comment:
Most of these duplicate the service groups.  It seems it would be better to split these into common ones, and service & slice specific ones?